### ISSUE TITLE
Auto Save and Auto Upload

### ISSUE THREAD

**Issue by madprops** [flameshot-org/flameshot#546]
Hi, very cool application.
Wanted to suggest a usage pattern I used a lot on Sharex on Windows.

Basically whenever I felt the urge to capture a region, it was 90% of the time to share it with other people through a link. So I hit a shortcut, quickly select a region and BOOM it was quickly uploaded to Imgur (url copied to clipboard) and saved in a local directory, without needing to hit any button.

I'm already mapping "flameshot gui" command to a shortcut, so I'm half way there. What I'm missing is more command line arguments to tell it to automatically upload to Imgur and/or save to a local directory. With enough command line arguments I can build my customized scripts.

When uploading to Imgur I want to copy the url to the clipboard pretty much 100% of the time.

So basically what I'm asking for is a way to configure a quicker way to upload/save, without having to press buttons. Though annotations can be useful sometimes, what Sharex does is provide buttons as soon as the screen 'freezes' instead of only when the region is selected. And when the region is selected it does all the actions automatically.

Thanks.

**Comment by ElijahLynn** [comment]
Cross linking similar issues:
* https://github.com/lupoDharkael/flameshot/issues/546
* https://github.com/lupoDharkael/flameshot/issues/394
* https://github.com/lupoDharkael/flameshot/issues/73
* https://github.com/lupoDharkael/flameshot/issues/734

**Comment by borgmanJeremy** [comment]
#516 is now merged for the 0.8 release.
